src/main/java/goodgenerator/blocks/tileEntity/base/LargeFusionComputer.java

```diff
@@ -388,101 +399,81 @@ public void onMachineBlockUpdate() {
         }
         if (tierOverclock() == 8) {
-            return (mStartEnergy < 160000000) ? 8
-                    : ((mStartEnergy < 320000000) ? 4 : (mStartEnergy < 640000000) ? 2 : 1);
+            return (mStartEnergy <= 160000000) ? 3
+                    : ((mStartEnergy <= 320000000) ? 2 : (mStartEnergy <= 640000000) ? 1 : 0);
         }
-        return (mStartEnergy < 160000000) ? 16
-                : ((mStartEnergy < 320000000) ? 8
-                        : ((mStartEnergy < 640000000) ? 4 : (mStartEnergy < 1280000000) ? 2 : 1));
+        return (mStartEnergy <= 160000000) ? 4
+                : ((mStartEnergy <= 320000000) ? 3
+                        : ((mStartEnergy <= 640000000) ? 2 : (mStartEnergy <= 1280000000) ? 1 : 0));
     }
 
-    @Override
-    public @NotNull CheckRecipeResult checkProcessing_EM() {
-
-        ArrayList<FluidStack> tFluidList = getStoredFluids();
+    public GT_Recipe.GT_Recipe_Map getRecipeMap() {
+        return GT_Recipe.GT_Recipe_Map.sFusionRecipes;
+    }
 
-        if (tFluidList.size() > 1) {
-            FluidStack[] tFluids = tFluidList.toArray(new FluidStack[0]);
-            GT_Recipe tRecipe = GT_Recipe.GT_Recipe_Map.sFusionRecipes
-                    .findRecipe(this.getBaseMetaTileEntity(), this.mLastRecipe, false, Integer.MAX_VALUE, tFluids);
+    @Override
+    protected ProcessingLogic createProcessingLogic() {
+        return new ProcessingLogic() {
+
+            @NotNull
+            @Override
+            protected GT_ParallelHelper createParallelHelper(@NotNull GT_Recipe recipe) {
+                // When the fusion first loads and is still processing, it does the recipe check without consuming.
+                if (mRunningOnLoad) {
+                    return new GT_ParallelHelper().setRecipe(recipe).setItemInputs(inputItems)
+                            .setFluidInputs(inputFluids).setAvailableEUt(availableVoltage * availableAmperage)
+                            .setMachine(machine, protectItems, protectFluids)
+                            .setRecipeLocked(recipeLockableMachine, isRecipeLocked).setMaxParallel(maxParallel)
+                            .enableBatchMode(batchSize).enableOutputCalculation();
+                }
+                return super.createParallelHelper(recipe);
+            }
 
-            if ((tRecipe == null && !mRunningOnLoad) || (tRecipe != null && (maxEUStore() < tRecipe.mSpecialValue))) {
-                turnCasingActive(false);
-                this.mLastRecipe = null;
-                return CheckRecipeResultRegistry.NO_RECIPE;
+            @NotNull
+            @Override
+            protected GT_OverclockCalculator createOverclockCalculator(@NotNull GT_Recipe recipe,
+                    @NotNull GT_ParallelHelper helper) {
+                return super.createOverclockCalculator(recipe, helper)
+                        .limitOverclockCount(overclock(recipe.mSpecialValue));
```

Comment:
Any reason you aren't setting the OC to be 4/4 when its an mk4/5?

Reply:
I set it in the PP base class.

Reply:
missed that one cus i only saw red there :trollface: